standardization/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -128,45 +130,85 @@ object StandardizationJob {
     spark
   }
 
-  private def readerFormatSpecific(dfReader: DataFrameReader, cmd: CmdConfig, dataset: Dataset): DataFrameReader = {
+  def getFormatSpecificReader(dfReader: DataFrameReader, cmd: CmdConfig, dataset: Dataset): DataFrameReader = {
     // applying format specific options
-    val dfReaderWithOptions = {
-      val dfr1 = if (cmd.rowTag.isDefined) dfReader.option("rowTag", cmd.rowTag.get) else dfReader
-      val dfr2 = if (cmd.csvDelimiter.isDefined) dfr1.option("delimiter", cmd.csvDelimiter.get) else dfr1
-      val dfr3 = if (cmd.csvHeader.isDefined) dfr2.option("header", cmd.csvHeader.get) else dfr2
-      val dfr4 = if (cmd.rawFormat.equalsIgnoreCase("cobol")) applyCobolOptions(dfr3, cmd, dataset) else dfr3
-      dfr4
-    }
-    if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
-      val dfWithFixedWithOptions = if (cmd.fixedWidthTrimValues.get) {
-        dfReaderWithOptions.option("trimValues", "true")
-      } else {
-        dfReaderWithOptions
+    val options = getCobolOptions(cmd, dataset) ++
+      getGenericOptions(cmd) ++
+      getXmlOptions(cmd) ++
+      getCsvOptions(cmd) ++
+      getFixedWidthOptions(cmd)
+
+    // Applying all the options
+    options.foldLeft(dfReader) { (df, optionPair) =>
+      optionPair match {
+        case (key, Some(value)) =>
+          value match {
+            case s: String => df.option(key, s)
+            case b: Boolean => df.option(key, b)
+            case l: Long => df.option(key, l)
+            case d: Double => df.option(key, d)
+          }
+        case (_, None) => df
       }
-      dfWithFixedWithOptions
+    }
+  }
+
+  private def getGenericOptions(cmd: CmdConfig): HashMap[String,Option[Any]] = {
+    HashMap("charset" -> cmd.charset)
+  }
+
+  private def getXmlOptions(cmd: CmdConfig): HashMap[String,Option[Any]] = {
+    if (cmd.rawFormat.equalsIgnoreCase("xml")) {
+      HashMap("rowtag" -> cmd.rowTag)
```

Comment:
The default value for rowTag is `None` so isn't this if/else redundant when you have line logic on line 143- 151? + CmdConfig has these checks on its own. Same for others.

Reply:
Yeah, it is redundant. But I wanted for format-specific option getters to follow a certain pattern. So when new options are added the developer should not worry about this.